partition_handler_dbus_service/src/main.rs

```diff
@@ -0,0 +1,127 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use clap::Parser;
+use std::process::Command;
+use zbus::{dbus_interface, ConnectionBuilder};
+
+#[derive(Debug, Parser)]
+struct Cli {
+    #[clap(short, long, required = true)]
+    partition_path: String,
+}
+
+struct PartitionHandler {
+    partition_path: String,
+}
+
+#[dbus_interface(name = "io.edgehog.PartitionHandler1")]
+impl PartitionHandler {
+    // Get boot consecutive fails
+    fn get_consecutive_fail_boot(&self) -> String {
+        self.get_value_for_key_from_info(&self.get_info_stdout(), "Consecutive fail boot")
+    }
+
+    // Get fail counter state
+    fn get_boot_fail_counter_state(&self) -> String {
+        self.get_value_for_key_from_info(&self.get_info_stdout(), "Boot fail counter state")
+    }
+
+    fn get_value_for_key_from_info(&self, stdout: &str, key: &str) -> String {
```

Comment:
Is this function exposed like the others via DBUS-API?maybe it should be something like a private function